<launch>
    <arg name="odom_frame_id"   default="volta_1_tf/odom" />
    <arg name="base_frame_id"   default="volta_1_tf/base_link" />
    <arg name="global_frame_id" default="map" />
    <arg name="laser_topic" default="/volta_1/scan_filtered" />

    <arg name="cmd_vel_topic" default="/volta_1/cmd_vel" />
    <arg name="odom_topic" default="/volta_1/odom" />
    <arg name="move_forward_only" default="false"/>

    <arg name="base_global" default="global_planner/GlobalPlanner"/>
    <arg name="base_local" default="dwa_local_planner/DWAPlannerROS"/>

    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen"> 
        <param name="base_global_planner" value="$(arg base_global)"/>
        <param name="base_local_planner" value="$(arg base_local)"/>
        <param name="global_costmap/global_frame" value="$(arg global_frame_id)" />
        <param name="global_costmap/robot_base_frame" value="$(arg base_frame_id)" />
        <param name="local_costmap/global_frame" value="$(arg global_frame_id)" />
        <param name="local_costmap/robot_base_frame" value="$(arg base_frame_id)" />
        <param name="DWAPlannerROS/global_frame_id" value="$(arg odom_frame_id)" />
        
        <rosparam file="$(find navigation)/config/volta_1/global_planner.yaml" command="load"/>
        <rosparam file="$(find navigation)/config/volta_1/local_planner.yaml" command="load"/>
        <rosparam file="$(find navigation)/config/volta_1/costmap.yaml" command="load" ns="global_costmap" />
        <rosparam file="$(find navigation)/config/volta_1/costmap.yaml" command="load" ns="local_costmap" />
        <rosparam file="$(find navigation)/config/volta_1/costmap_local.yaml" command="load" ns="local_costmap" />
        <rosparam file="$(find navigation)/config/volta_1/costmap_global.yaml" command="load" ns="global_costmap"/>
        
        <remap from="cmd_vel"                                                                       to="$(arg cmd_vel_topic)"/>
        <remap from="odom"                                                                          to="$(arg odom_topic)"/>
        <remap from="scan"                                                                          to="$(arg laser_topic)"/>
        <remap from="map"                                                                           to="/map"/>
    </node>

</launch>